### PR TITLE
Diagnostic HUD

### PR DESCRIPTION
**Pull request by DamianX**
Status bar sprites stolen from TG, the glasses icon is a codersprite made by changing the colors of the health scanner HUD to something orange-ish.

The only features for now are health and power cell status bars for borgs and mechs, I plan on introducing feature-creep and maybe give it to the AI/observers/protolathe later, but I want to refactor HUD code first.

Edited wiki: http://ss13.moe/wiki/index.php?title=Guide_to_Robotics&diff=11508&oldid=11469

:cl:
 * rscadd: Added the diagnostic HUD: lets the user see integrity and power cell charge for cyborgs and mechs. A couple of them can be found in the roboticist's locker.

### CODE REVIEW

File: __DEFINES/setup.dm [hunk 1]
```diff
@@ -864,15 +864,17 @@ SEE_PIXELS	256
...
-#define HEALTH_HUD          1 // a simple line rounding the mob's number health
-#define STATUS_HUD          2 // alive, dead, diseased, etc.
-#define ID_HUD              3 // the job asigned to your ID
-#define WANTED_HUD          4 // wanted, released, parroled, security status
-#define IMPLOYAL_HUD		5 // loyality implant
-#define IMPCHEM_HUD		    6 // chemical implant
-#define IMPTRACK_HUD		7 // tracking implant
-#define SPECIALROLE_HUD 	8 // AntagHUD image
-#define STATUS_HUD_OOC		9 // STATUS_HUD without virus db check for someone being ill.
+#define HEALTH_HUD          "health" // a simple line rounding the mob's number health
```
Comment (DamianX) [inline __DEFINES/setup.dm:867]: `hud_updateflag` is broken with these changes but it's not actually used so I'm not too concerned about it.

File: code/__HELPERS/mobs.dm [hunk 2]
```diff
@@ -153,6 +153,40 @@ proc/RoundHealth(health)
 			return "health-100"
 	return "0"
 
+/proc/cyborg_health_to_icon_state(var/health)
+	switch(health)
```
Comment (jknpj) [inline code/__HELPERS/mobs.dm]: Consider using percentage instead of fixed values

**Comment by DrSnips** [comment]
You might want to consider adding them as a protolathe/exofab design or something since its science equipment. They could also use research values at the bare minimum even if not added to the protolathe/exofab.

**Comment by DamianX** [comment]
I agree, as I've said in the OP

**Review by PJB3005 — APPROVED** [review]

**Pull request merged**